src/include/function/comparison/vector_comparison_operations.h

```diff
@@ -42,44 +66,44 @@ class VectorComparisonOperations : public VectorOperations {
         common::PhysicalTypeID leftType, common::PhysicalTypeID rightType, scalar_exec_func& func) {
-        assert(leftType == rightType);
         switch (leftType) {
         case common::PhysicalTypeID::INT64: {
-            func = BinaryExecFunction<int64_t, int64_t, uint8_t, FUNC>;
-            return;
-        }
+            func = BinaryComparisonExecFunction<int64_t, int64_t, uint8_t, FUNC>;
+        } break;
         case common::PhysicalTypeID::INT32: {
-            func = BinaryExecFunction<int32_t, int32_t, uint8_t, FUNC>;
-            return;
-        }
+            func = BinaryComparisonExecFunction<int32_t, int32_t, uint8_t, FUNC>;
+        } break;
         case common::PhysicalTypeID::INT16: {
-            func = BinaryExecFunction<int16_t, int16_t, uint8_t, FUNC>;
-            return;
-        }
+            func = BinaryComparisonExecFunction<int16_t, int16_t, uint8_t, FUNC>;
+        } break;
         case common::PhysicalTypeID::DOUBLE: {
-            func = BinaryExecFunction<double, double, uint8_t, FUNC>;
-            return;
-        }
+            func = BinaryComparisonExecFunction<double, double, uint8_t, FUNC>;
+        } break;
         case common::PhysicalTypeID::FLOAT: {
-            func = BinaryExecFunction<float, float, uint8_t, FUNC>;
-            return;
-        }
+            func = BinaryComparisonExecFunction<float, float, uint8_t, FUNC>;
+        } break;
         case common::PhysicalTypeID::BOOL: {
-            func = BinaryExecFunction<uint8_t, uint8_t, uint8_t, FUNC>;
-            return;
-        }
+            func = BinaryComparisonExecFunction<uint8_t, uint8_t, uint8_t, FUNC>;
+        } break;
         case common::PhysicalTypeID::STRING: {
-            func = BinaryExecFunction<common::ku_string_t, common::ku_string_t, uint8_t, FUNC>;
-            return;
-        }
+            func = BinaryComparisonExecFunction<common::ku_string_t, common::ku_string_t, uint8_t,
+                FUNC>;
+        } break;
         case common::PhysicalTypeID::INTERNAL_ID: {
-            func = BinaryExecFunction<common::nodeID_t, common::nodeID_t, uint8_t, FUNC>;
-            return;
-        }
+            func = BinaryComparisonExecFunction<common::nodeID_t, common::nodeID_t, uint8_t, FUNC>;
+        } break;
         case common::PhysicalTypeID::INTERVAL: {
-            func = BinaryExecFunction<common::interval_t, common::interval_t, uint8_t, FUNC>;
-            return;
-        }
+            func =
+                BinaryComparisonExecFunction<common::interval_t, common::interval_t, uint8_t, FUNC>;
+        } break;
+        case common::PhysicalTypeID::VAR_LIST: {
+            func = BinaryComparisonExecFunction<common::list_entry_t, common::list_entry_t, uint8_t,
+                FUNC>;
+        } break;
+        case common::PhysicalTypeID::STRUCT: {
+            func = BinaryComparisonExecFunction<common::struct_entry_t, common::struct_entry_t,
+                uint8_t, FUNC>;
+        } break;
```

Comment:
If you are using break here, make sure to change getSelectFunc to use the same style